lib/signal-manager.js

```diff
@@ -0,0 +1,44 @@
+const runningProcs = new Set()
+
+const forwardedSignals = [
+  'SIGINT',
+  'SIGTERM'
+]
+
+function handleSignal (signal) {
+  for (const proc of runningProcs) {
+    proc.kill(signal)
```

Comment:
So, if we SIGTERM the npm process, it passes SIGTERM to the child, but then doesn't _itself_ die from the signal?  Seems like we ought to add a listener and if the signal is fatal to the child proc, also die in the same way?  (I suppose, in all such cases, we _will_ terminate with an error, since it'd be a `npm run-script` command where the process exits in error?)

Reply:
It looks like in v6, we have some complicated logic where we pass the SIGINT to the child, then the next SIGINT we pass to the child as SIGTERM, then we kill it with SIGKILL.  That's weird.

Maybe we should just make these `process.on()` instead of `process.once()`?  If the child dies, then we exit in error, but otherwise, we let you ^C to your hearts' content.

Reply:
we don't manually do anything to die on our own because the use case here is, as you mentioned, a child process run by `npm run-script`. forwarding the signal to the child and then bubbling the exit code from the child back out to the parent is sufficient since the child is the only thing keeping `npm` itself running. child dies, so does npm immediately after.

i agree that the flow of SIGINT -> SIGTERM -> SIGKILL is weird, my idea here with using `process.once()` instead of `process.on()` is to effectively make the first SIGINT forward to the child and if that fails for some reason (i.e. the child is frozen) then the second SIGINT will go to npm itself which is a bit more brute force. without that escape hatch if a child executed by `npm run-script` has a SIGINT handler that doesn't actually exit for some reason we would be stuck requiring the user to resort to a SIGTERM or SIGKILL themselves.

i'm definitely open to switching this to a `process.on()` if we decide it's more intuitive to just always forward the signals any time a child is running and force the user to escalate the signal they send on their own.